### PR TITLE
Tflogging juju client

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description

Updating logging package for internal/juju/*. And version location in makefile.

Fixes: 

## Type of change

- Maintenance work (repository related, like Github actions, or revving the Go version, etc.)

## QA steps

To setup logging: 
```
export TF_LOG_PROVIDER=TRACE ; export TF_LOG_PATH=./terraform.log
```

Check log file for `@module:juju.client`

### CODE REVIEW

File: internal/juju/models.go [hunk 1]
```diff
@@ -150,8 +149,7 @@ func (c *modelsClient) GetModelByName(name string) (*params.ModelInfo, error) {
 
 	modelInfo := results[0].Result
 
-	log.Printf("[DEBUG] Reading model: %s, %+v", name, modelInfo)
-
+	c.Tracef(fmt.Sprintf("Reading model: %s, %+v", name, modelInfo))
```
Comment (cderici) [inline internal/juju/models.go:152]: I think "Reading" is misleading verb here, as it means something in the provider context.

```suggestion
	c.Tracef(fmt.Sprintf("Retrieving model info from Juju api: %s, %+v", name, modelInfo))
```

File: internal/juju/sshKeys.go [hunk 2]
```diff
@@ -128,7 +127,7 @@ func (c *sshKeysClient) DeleteSSHKey(input *DeleteSSHKeyInput) error {
 		k := returnedKeys[0].Result[0]
 		if input.User == utils.GetUserFromSSHKey(k) {
 			// This is the latest key, do not remove it
-			log.Warn().Msgf("ssh key from user %s is the last one and will not be removed", input.User)
+			c.Warnf(fmt.Sprintf("ssh key from user %s is the last one and will not be removed", input.User))
```
Comment (cderici) [inline internal/juju/sshKeys.go:130]: Is the `fmt.Sprintf` redundant here, can we just: ?

```suggestion
			c.Warnf("ssh key from user %s is the last one and will not be removed", input.User)
```

File: internal/juju/client.go [hunk 3]
```diff
@@ -83,8 +86,51 @@ func (cf *ConnectionFactory) GetConnection(model *string) (api.Connection, error
...
+//	@module=juju.client
+const LogJujuClient = "client"
+
+// TODO hml 04-Aug-2023
+// Investigate if the context from terraform can be passed in
+// and used with tflog here, for now, use context.Background.
+
+func (cf *ConnectionFactory) Debugf(msg string, additionalFields ...map[string]interface{}) {
+	if cf.subCtx == nil {
+		cf.subCtx = tflog.NewSubsystem(context.Background(), LogJujuClient)
```
Comment (cderici) [inline internal/juju/client.go:108]: Should we use `context.TODO` here, as it's  not super clear which context it should be? might be more appropriate, though `Background` makes sense too, I'm not entirely sure, just wanted to point out

**Review by cderici — CHANGES_REQUESTED** [review]
A couple of nit picks, and one change, and it'll be good to go 👍

File: internal/juju/sshKeys.go (hunk 2 above)
Comment (hmlanigan) [reply internal/juju/sshKeys.go:130]: no, the 2nd argument to warn is a map[string]interface.

File: internal/juju/client.go (hunk 3 above)
Comment (hmlanigan) [reply internal/juju/client.go:108]: It's highly unlikely we'll get a context here, nor do I believe it's useful to pipe one in from the provider code just for logging. However a non nil context is required.

File: Makefile [hunk 4]
```diff
@@ -13,7 +13,7 @@ go-install:
 GOOS=$(shell go env GOOS)
 GOARCH=$(shell go env GOARCH)
 GOPATH=$(shell go env GOPATH)
-VERSION=0.8.0
+VERSION=0.9.0
```
[the remark this answers is not in the document]
Comment (hmlanigan) [reply Makefile:16]: This is for testing purposes only. To ensure the version we use in testing, doesn't match a released version. In many ways this is a hack, it should be a non released version. Maybe 0.100.0?

[the remark this answers is not in the document]
Comment (cderici) [reply Makefile:16]: Yeah I think that's better than 0.9.0, just to save ourselves a headache later on

**Review by cderici — APPROVED** [review]
There's only the version change thing, otherwise LGTM 👍

**Pull request merged**